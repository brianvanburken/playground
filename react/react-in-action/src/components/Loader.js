import React from 'react';

/**
 * See https://www.npmjs.com/package/loaders.css for more info
 * @method Loader
 */
const Loader = () => (
    <div className="loader">
        <div className="ball-pulse-sync">
            <div />
            <div />
            <div />
        </div>
    </div>
);

export default Loader;
