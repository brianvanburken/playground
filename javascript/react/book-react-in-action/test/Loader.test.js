import React from 'react';
import Loader from '../src/components/Loader';
import renderer from 'react-test-renderer';

test('Loader', () => {
    const component = renderer.create(<Loader />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
