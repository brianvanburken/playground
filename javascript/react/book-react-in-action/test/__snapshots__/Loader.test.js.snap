// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader 1`] = `
<div
  className="loader"
>
  <div
    className="ball-pulse-sync"
  >
    <div />
    <div />
    <div />
  </div>
</div>
`;
